x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/Match.java

```diff
@@ -67,14 +71,22 @@ private static Match readFrom(StreamInput in) throws IOException {
         Source source = Source.readFrom((PlanStreamInput) in);
         Expression field = in.readNamedWriteable(Expression.class);
         Expression query = in.readNamedWriteable(Expression.class);
-        return new Match(source, field, query);
+        Match match = new Match(source, field, query);
+
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_MATCH_WITH_SEMANTIC_TEXT)) {
```

Comment:
Should be fixed now - `inferenceResults` is now part of the constructor.
However there are 2 constructors:
`Match(source, field, query)` this is used by `EsqlFunctionRegistry`.
`Match(source, field, query, inferenceResults)` this is used for deserialization and in a static method `Match.newWithInferenceResults(match, inferenceResults)`. the static method is used by `InferenceResolver`.
The reason why this second constructor is private is because if it was public when I tested we would get a compilation error for `EsqlFunctionRegistry` because we would have 2 constructors:

https://github.com/elastic/elasticsearch/blob/ab604ada78d779a18b82465d51829006540ce546/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java#L404

If we made it just keep this constructor `Match(source, field, query, inferenceResults)` this would require more changes, changing the `FunctionDefinition` and `EsqlFunctionRegistry` (similar to how we did to account that some functions can receive `Configuration`).
I did not see the benefit of that given that AFAIK `match` should be the only function that receives `inferenceResults`.